File: app/models/module.py
from pydantic import BaseModel
from typing import Optional
from uuid import UUID
from datetime import datetime
from decimal import Decimal

class ModuleBase(BaseModel):
    code: str
    name: str
    description: Optional[str] = None
    base_price: Decimal

class ModuleResponse(ModuleBase):
    module_id: UUID
    is_active: bool
    created_at: datetime

    class Config:
        from_attributes = True

class TenantModuleUpdate(BaseModel):
    module_id: UUID
    is_enabled: bool
    price_override: Optional[Decimal] = None

class TenantModuleResponse(BaseModel):
    id: UUID
    tenant_id: UUID
    module: ModuleResponse
    status: str
    price_override: Optional[Decimal]
    enabled_at: datetime
    
    class Config:
        from_attributes = True

File: admin-dashboard/types/module.ts
export interface Module {
    module_id: string;
    code: string;
    name: string;
    description: string;
    base_price: string; // Decimal comes as string from JSON usually
    is_active: boolean;
    created_at: string;
}

export interface TenantModule {
    id: string;
    tenant_id: string;
    module: Module;
    status: 'active' | 'disabled';
    price_override?: string;
    enabled_at: string;
}

export interface TenantModuleUpdate {
    module_id: string;
    is_enabled: boolean;
    price_override?: number;
}
